File: dsa-problems/Sorting/solutions/javascript/SRT-001-partial-selection-sort-stats.js
class Solution {
  partialSelectionSort(arr, k) {
    const n = arr.length;
    for (let i = 0; i < k; i++) {
      let minIndex = i;
      for (let j = i + 1; j < n; j++) {
        if (arr[j] < arr[minIndex]) {
          minIndex = j;
        }
      }
      [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
    }
    return arr;
  }
}

File: dsa-problems/Sorting/solutions/java/SRT-001-partial-selection-sort-stats.java
import java.util.*;

class Solution {
    public int[] partialSelectionSort(int[] arr, int k) {
        int n = arr.length;
        for (int i = 0; i < k; i++) {
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            // Swap
            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
        return arr;
    }
}
